drivers/video/fbdev/bcm2708_fb.c

```diff
@@ -869,101 +1000,172 @@ static int bcm2708_fb_probe(struct platform_device *dev)
 {
 	struct device_node *fw_np;
 	struct rpi_firmware *fw;
-	struct bcm2708_fb *fb;
-	int ret;
+	int ret, i;
+	u32 num_displays;
+	struct bcm2708_fb_dev *fbdev;
+	struct { u32 base, length; } gpu_mem;
+
+	fbdev = kzalloc(sizeof(*fbdev), GFP_KERNEL);
```

Comment:
Recommendation is to use devm_kzalloc(&pdev->dev, sizeof(*fbdev), GFP_KERNEL) to allow the kernel to manage the lifespan of the memory allocation. It'll be freed automagically when the last reference on the device is released. No need for kfree or equivalent anywhere.